core/domain/user_services.py

```diff
@@ -111,7 +119,27 @@ def require_valid_username(cls, username):
             # Admin usernames are reserved for admins. Note that 'admin'
             # itself is already in use for the demo exploration.
             raise utils.ValidationError('This username is not available.')
-
+    @classmethod
+    def fetch_gravatar(cls, email):
+        """Gets autogenerated avatar for a given email.
+
+        Returns Default picture if there is an error.
+        """
+        base_url = "http://www.gravatar.com/avatar/"
+        avatar_url = base_url + hashlib.md5(email.lower()).hexdigest() + "?"
+        avatar_url += urllib.urlencode({'d':'retro', 's':str(AVATAR_SIZE)})
+        try:
+            request = urllib2.Request(avatar_url)
+            response = urllib2.urlopen(request)
+            if response.getcode() == 200:
+                content_type = response.headers.getheader('content-type')
```

Comment:
Yeah we can , by appending .png at the end of url we'll always receive .png fiile.